### PR TITLE
Post link polish 2

### PR DESCRIPTION
**Pull request by cindywu**
- [x] do not italicize text in tooltip if highlighted text is italics
- [x] polish styles for post link search form 
- [x] fix suggestionslist box width
- [x] keep italics in post title for postlink tooltip
- [x] fix having to refresh show title of post link
- [x] source url for existing posts
- [x] styles for post link show tooltip
- [x] renames all instances of postlink(s) with citation(s)

### CODE REVIEW

File: app/javascript/components/editor-config/citations.js [hunk 1]
```diff
@@ -89,20 +91,23 @@ export const addPostLink = function (state, dispatch, view) {
 
     const handleClose = () => ReactDOM.unmountComponentAtNode(root)
 
-    const handleNewPostLink = (payload) => {
-      console.log('in handleNewPostLink this is the payload', payload)
+    // TODO: refactor
```
Comment (dml-1) [inline app/javascript/components/editor-config/citations.js:94]: what does this comment mean?

File: app/javascript/components/editor-config/citations.js [hunk 2]
```diff
@@ -288,27 +293,34 @@ function renderPostLinks(postLinks, dispatch, state) {
...
   const handleDelete = () => {
     dispatch(
-      state.tr.setMeta(postLinkPlugin, { type: 'deletePostLink', postLink })
+      state.tr.setMeta(citationPlugin, { type: 'deleteCitation', citation })
     )
   }
 
   var { currentUser } = store.getState()
+
+  const sanitizedTitle = parse(
```
Comment (dml-1) [inline app/javascript/components/editor-config/citations.js:307]: Please replace this with the `sanitizeHtml` library we already use, e.g. on `PostPage`, and remove this library from our packages.

File: app/javascript/components/editor-config/citations.js (hunk 1 above)
Comment (cindywu) [reply app/javascript/components/editor-config/citations.js:94]: There is repetitive code in `handleNewCitation` that can be made concise. Right now I am not sure how to refactor while preserving the Promise.

I removed the TODO.

File: app/javascript/components/editor-config/citations.js (hunk 2 above)
Comment (cindywu) [reply app/javascript/components/editor-config/citations.js:307]: `sanitizeHtml` doesn't solve the issue, so keeping `html-react-parser`.

**Pull request merged**